linode/manager
# test: [M3-8948] Added cypress component test for firewall inbound and outbound rules for mouse drag and drop

## Description 📝

This PR is part of adding test for firewall drag and drop with mouse and keyboard. (M3-8862)

In this PR cypress component test is added for firewall inbound and outbound rules for mouse drag and drop

## Changes  🔄

Cypress Test for Linode Firewall Inbound & Outbound Rules Table Mouse Drag and Drop Functionality

![image](https://github.com/user-attachments/assets/f84e7285-71a4-496c-9b98-deb82a774593)

In this PR, I implemented and tested the drag-and-drop functionality in Cypress. Below is the approach I followed:

1. Using cypress trigger events: Attempted to simulate drag-and-drop actions using native Cypress trigger events, but this approach did not work as expected.
2. Using cypress-real-events:Tried using cypress-real-events to replicate drag-and-drop functionality. However, it also failed to work for this specific use case.
3. Using 4teamwork/cypress-drag-drop:Adopted the [4teamwork/cypress-drag-drop](https://github.com/4teamwork/cypress-drag-drop) library, which successfully enabled the drag-and-drop functionality.

**This implementation uses the 4teamwork/cypress-drag-drop plugin to ensure reliable and consistent test execution for drag-and-drop scenarios.**

## Jira
- M3-8948

## How to test 🧪
`yarn cy:component:run -s "cypress/component/features/firewalls/firewall-rule-table.spec.tsx"`

### Verification steps

When test executed using this command  `yarn cy:component:run -s "cypress/component/features/firewalls/firewall-rule-table.spec.tsx"`all tests should pass successfully and you should see new test added as per below screenshot 

![image](https://github.com/user-attachments/assets/69709061-185c-4a0a-96f2-93f5c51a758b)


## As an Author I have considered 🤔

*Check all that apply*
